contrib/win32/win32compat/misc.c

```diff
@@ -283,33 +283,49 @@ char*
  w32_fgets(char *str, int n, FILE *stream) {
 	HANDLE h = (HANDLE)_get_osfhandle(_fileno(stream));
 	wchar_t* str_w = NULL;
-	char *ret = NULL, *str_tmp = NULL;
+	char *ret = NULL, *str_tmp = NULL, *cp = NULL;
```

Comment:
No, according to https://linux.die.net/man/3/fgets, only return str on success; will return NULL on error